blinky-statik/src/main/java/org/spideruci/analysis/db/DatabaseWriter.java

```diff
@@ -0,0 +1,98 @@
+package org.spideruci.analysis.db;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.spideruci.analysis.statik.flow.data.SourceICFGNode;
+
+public class DatabaseWriter {
+	private static Connection c = null;
+	private static Map<SourceICFGNode, Integer> nodeIdMap = new HashMap<>();
+	private static int nodeId = 0; 
+
+	public DatabaseWriter(Connection c) {
+		this.c = c;
```

Comment:
is `c` an instance field or a static field? If it is an instance field, then you should remove the static keyword in its declaration above (remove `static` in line Line13: `private static Connection c = null;`). If it is a static field, then it should not be assigned in the constructor of DataBaseWriter, but should use a static method ... something like `public static DataBaseWriter init(Connection c)`.